truss/templates/control/control/endpoints.py

```diff
@@ -151,3 +152,36 @@ def _is_streaming_response(resp) -> bool:
         if header_name.lower() == "transfer-encoding" and value.lower() == "chunked":
             return True
     return False
+
+
+def _reroute_if_health_check(path: str) -> str:
+    if path == "/v1/models/model":
+        # Reroute health checks to the inference server's /v1/models/model/loaded endpoint
+        path = "/v1/models/model/loaded"
+    return path
+
+
+def _custom_wait_strategy(retry_state: RetryCallState) -> int:
+    # Apply exponential backoff only for ModelNotReady
+    if retry_state.outcome is not None and isinstance(
+        retry_state.outcome.exception(), ModelNotReady
+    ):
+        return min(2**retry_state.attempt_number, 10)
+    # Use fixed wait for other exception types
+    return 1
+
+
+def _custom_stop_strategy(retry_state: RetryCallState) -> bool:
+    # Stop after 10 attempts for ModelNotReady
```

Comment:
How long can this end up waiting and retrying?

Reply:
should be capped at 75 seconds from first attempt to the last attempt.

Reply:
I think this could be shorter. 75s seems long, and I'm not sure anything is going to wait that long for a response on the healthcheck. For example, the operator's retry timeout on this is 2s. The `wake` call might wait longer though

Reply:
did some digging and looks like we just call `GET /` on `wake`, which shouldn't take long. I'll go ahead and update the retries here to be once a second for 10 attempts